project/src/main/java/com/google/sps/servlets/NewsService.java

```diff
@@ -15,35 +15,59 @@
 package com.google.sps.servlets;
 
 import okhttp3.*;
+import com.rometools.rome.feed.synd.SyndFeed;
+import com.rometools.rome.feed.synd.SyndEntry;
+import com.rometools.rome.io.SyndFeedInput;
+import com.rometools.rome.io.XmlReader;
 
-import java.io.IOException;
+import java.net.*;
+import java.util.*;
 
 /** Service class that delivers news under a requested topic. */
 public class NewsService {
   // one instance, reuse
   private final OkHttpClient httpClient = new OkHttpClient();
 
   // Retrieves articles from the Google News RSS Feed using the topic parameter as the search query
-  // The Topic parameter allows for spaces and non alphanumeric characters, but null is an invalid argument
-  public String retrieveNewArticles(String topic) {
+  // The Topic parameter allows for spaces and non alphanumeric characters, but null and "" is an invalid argument
+  public String retrieveNewArticles(String topic, int numArticles) {
     String url = String.format("https://news.google.com/rss/search?q=%s", topic);
     String articleXml;
     try {
-      articleXml = httpRequest(url);
-    } catch(IOException ioexeption) {
-      articleXml = "IOException occured when requesting articles from the Google RSS News Feed";
+      articleXml = cleanSyndFeed(getSyndFeed(url), numArticles).toString();
+    } catch(Exception exception) {
+      articleXml = "Exception occured when requesting articles from the Google RSS News Feed";
     }
 
     return articleXml;
   }
 
-  private String httpRequest(String url) throws IOException {
-    Request request = new Request.Builder().url(url).build();
+  private SyndFeed getSyndFeed(String url) throws Exception{
+    // Encode URL so that it can include spaces
+    String encodedUrl = url.replaceAll(" ", "%20");
+    // Builiding SyndFeed
+    URL feedSource = new URL(encodedUrl);
+    SyndFeedInput input = new SyndFeedInput();
+    SyndFeed feed = input.build(new XmlReader(feedSource));
 
-    try (Response response = httpClient.newCall(request).execute()) {
-      if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
-      // Get response body
-      return response.body().string();
+    return feed;
+  }
+
+  private List<Article> cleanSyndFeed(SyndFeed syndFeed, int numArticles) {
+    // We extract the entries into a managable list
```

Comment:
nit: *manageable

Reply:
lol, my spelling just went out of the window for this PR.